src/engine_core/rendering/src/Shared/RenderableNode.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <glm/mat4x4.hpp>
+#include <vector>
+#include <memory>
+#include "IRenderable.hpp"
+namespace Hush {
+	/// @brief Common renderable node for scenes with multiple children to render
+	/// the Draw function *MUST* be called (recursed down) for every implementation
+	/// PENDING: Unite this into one renderableNode implementation IF AND ONLY IF this becomes a bottleneck
+	class RenderableNode : public IRenderable {
+
+	public:
+		void RefreshTransform(const glm::mat4& parentMatrix) {
+			this->m_worldTransform = parentMatrix * this->m_localTransform;
+			for (std::shared_ptr<RenderableNode>& child : this->m_children) {
+				child->RefreshTransform(this->m_worldTransform);
+			}
+		}
+	
+		void Draw(const glm::mat4& topMatrix, void* drawContext) override {
+			for (std::shared_ptr<RenderableNode>& child : this->m_children) {
+				child->Draw(topMatrix, drawContext);
+			}
+		}
+
+		void SetLocalTransform(const glm::mat4& localTransform) {
+			this->m_localTransform = localTransform;
+		}
+
+		void SetWorldTransform(const glm::mat4& worldTransform) {
+			this->m_worldTransform = worldTransform;
+		}
+
+		const glm::mat4& GetLocalTransform() const noexcept {
+			return this->m_localTransform;
+		}
+
+		const glm::mat4& GetWorldTransform() const noexcept {
+			return this->m_worldTransform;
+		}
+
+	protected:
+		std::weak_ptr<RenderableNode> m_parent;
```

Comment:
With the previous approach, tracking their parents should not be needed in this component, it only contains data (i.e. the local and world transform).